### PR TITLE
debian: Ensure existence of /opt/cni/bin before installing kubernetes…

### LINKED ISSUE #64018: debian: kubernetes-cni.deb does not create required directories
/kind bug

**What happened**:

If you start with a minimal Debian image and attempt to install kubernetes-cni.deb as an artifact
from a Bazel build, it will fail with:

```
dpkg: error processing archive kubernetes-cni.deb (--install):
 error creating directory '/opt/cni/bin': No such file or directory
```

This has been a known issue in the past, and there are workarounds for CI in place, but it's
still unexpected behaviour (https://github.com/kubernetes/kubernetes/issues/59762#issuecomment-366400634)

**What you expected to happen**:

kubernetes-ci.deb installs.

**How to reproduce it (as minimally and precisely as possible)**:

``` shell
docker run -it -v <path to debs>:/mnt debian /bin/sh -c 'dpkg -i /mnt/kubernetes-cni.deb'
```

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): master
- Cloud provider or hardware configuration: n/a
- OS (e.g. from /etc/os-release): debian
- Kernel (e.g. `uname -a`): n/a
- Install tools: `dpkg`
- Others:

/sig cluster-lifecycle

### PR DESCRIPTION
**Pull request by randomvariable**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #64018 

**Special notes for your reviewer**:

 Only for bazel builds. rpms not affected

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by dims** [comment]
LGTM 👍

**Review by luxas — APPROVED** [review]
Thanks @randomvariable!

/lgtm
/approve
/milestone v1.11
/kind bug
/priority critical-urgent
/status approved-for-milestone
cc @timothysc @chuckha this is probably kinda important for our CI testing env depending on the bazel rpms
/hold
for @timothysc and @ixdy approval as well
/assign @timothysc @ixdy

File: build/debs/kubernetes-cni.preinst.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
```
Comment (ixdy) [inline build/debs/kubernetes-cni.preinst.sh]: does this file need to be executable? it doesn't look like it is right now.

File: build/debs/BUILD [hunk 2]
```diff
@@ -164,6 +164,7 @@ k8s_deb(
     description = """Kubernetes Packaging of CNI
 The Container Networking Interface tools for provisioning container networks.
 """,
+    preinst = "kubernetes-cni.preinst.sh",
```
Comment (timothysc) [inline build/debs/BUILD]: I'm not a debian packaging expert.  /cc @marcoceppi @castrojo 

With rpm.specs you own the directories in the spec.

**Comment by ixdy** [comment]
I think this might be the wrong fix, as I noted in https://github.com/kubernetes/kubernetes/issues/64018#issuecomment-396423137.

**Comment by randomvariable** [comment]
As per https://github.com/kubernetes/kubernetes/issues/64018#issuecomment-396423137, going to dig some more and see if it's resolvable without a workaround. Closing this until then.

**Pull request closed without merging**